hook!-full-game/src/Player/PassThroughDetector.gd

```diff
@@ -0,0 +1,16 @@
+extends Area2D
+
+onready var kinematic_body: KinematicBody2D = get_node(kinematic_body_path) as KinematicBody2D
+export var kinematic_body_path: NodePath = ".."
+
+func _ready() -> void:
+	connect("body_exited", self, "_on_body_exited")
+
+func _physics_process(delta: float) -> void:
+	if kinematic_body.is_on_floor() and not kinematic_body.get_collision_mask_bit(3):
+		kinematic_body.set_collision_mask_bit(3, true)
```

Comment:
I'd rename `kinematic_body` to `body` and add a constant for the bit mask. `PASS_THROUGH_LAYER` or something

Reply:
`body` is the default for the `_on_body_*` signals, might be confusing having an internal `body` var.

In any case, the `body` from the aignal callback should be renamed to `platform`?

Reply:
`_body`, `_actor`, etc. could work for the node's property.
In the signal, as you're using physics layers to only detect platforms, yes, you could rename it to platform. Or keep body

Reply:
I think I'll go for `_actor` sounds more explicit